### PR TITLE
Add support for symfony/stopwatch component

### PR DESCRIPTION
**Pull request by veteran29**
Basic support for profiling via symfony/stopwatch:
image

### CODE REVIEW

File: APM/PSRCommandLogger.php [hunk 1]
```diff
@@ -56,14 +61,24 @@ public function commandStarted(CommandStartedEvent $event)
             return;
         }
 
+        if ($this->stopwatch !== null) {
+            $this->stopwatch->start('mongodb', 'mongodb');
```
Comment (alcaeus) [inline APM/PSRCommandLogger.php]: I would suggest using the command name here to make the data more informative. Same in the corresponding `stop` calls.

Comment (veteran29) [reply APM/PSRCommandLogger.php]: Maybe with a `mongodb_` prefix?

It looks like this right now:
image

File: APM/PSRCommandLogger.php [hunk 2]
```diff
@@ -56,14 +61,24 @@ public function commandStarted(CommandStartedEvent $event)
             return;
         }
 
+        if ($this->stopwatch !== null) {
+            $this->stopwatch->start($event->getCommandName(), 'mongodb');
```
Comment (jmikola) [inline APM/PSRCommandLogger.php]: I think this is preferable to the earlier suggestion about prefixing the name with `mongodb_`, as the category argument will ensure events are grouped (even if the default UI only displays text for the command name itself by default).

It's interesting that `stop()` doesn't also take a category name. That makes me wonder how it determines that we're closing the same event and whether we could possibly conflict if some other handler used a generic event name (e.g. "find", "update"); however, I haven't worked with the component directly so perhaps this isn't a concern.

File: Resources/config/mongodb.xml [hunk 3]
```diff
@@ -138,6 +139,8 @@
 
         <service id="doctrine_mongodb.odm.command_logger" class="Doctrine\Bundle\MongoDBBundle\APM\PSRCommandLogger" public="false">
             <argument type="service" id="logger" on-invalid="null" />
+            <argument>%doctrine_mongodb.odm.command_logger.prefix%</argument>
```
Comment (jmikola) [inline Resources/config/mongodb.xml]: @alcaeus: If this is now part of the service definition, would it make sense to change the default value in the constructor signature to `null`?

File: APM/PSRCommandLogger.php (hunk 2 above)
Comment (alcaeus) [reply APM/PSRCommandLogger.php]: @jmikola the `stop` method always works on the last open section, not on a specific one. 🤷‍♂️

File: Resources/config/mongodb.xml (hunk 3 above)
Comment (alcaeus) [reply Resources/config/mongodb.xml]: I don't like using DI parameters for this, at least not to have this configurable to the outside. Instead, this should be configured using the configuration and then injected into the DI container by the extension.

**Pull request closed without merging**